openedx/core/djangoapps/user_api/api.py

```diff
@@ -796,8 +797,10 @@ def _add_honor_code_field(self, form_desc, required=True):
         Keyword Arguments:
             required (bool): Whether this field is required; defaults to True
         """
+
+        separate_honor_and_tos = self._is_field_visible("terms_of_service")
         # Separate terms of service and honor code checkboxes
-        if self._is_field_visible("terms_of_service"):
+        if separate_honor_and_tos:
```

Comment:
@dianakhuang: We mentioned renaming this to `separate_honor_and_terms` for consistency with other variables.

Reply:
Whoops, sorry. I merged already. The way I handled the `honor_code` variable was to add a hidden field to the form so that the value gets passed through that way.